extension/lib/bigcommerce/customer/shapeRequestError.js

```diff
@@ -0,0 +1,28 @@
+const BigCommerceRequestClientError = require('./request/ClientError')
+const BigCommerceRequestUnparseableMessageError = require('./request/UnparsableMessageError')
+
+module.exports = function shapeRequestError (err) {
+  // Try parsing the (potential) underlying api error
+  const errorMessageMatch = err.message.match(/({.+})/)
+  if (!errorMessageMatch) {
+    return err
+  }
+
+  let parsed
+  try {
+    parsed = JSON.parse(errorMessageMatch[1])
+  } catch (unparseable) {
+    return new BigCommerceRequestUnparseableMessageError('unable to parse')
+  }
+
```

Comment:
If you'd move everything below this line into the try block, you wouldn't have the problem of using the `parsed` variable in different scopes. Thus, you could define it with `const`.

Reply:
I think this is a matter of style, and there are some benefits to it that might not be obvious at the fist glance - for instance when we want to have everything inside try/catch block it is often the case that we have the need to throw a custom error - that will cause the error to be caught by the same try/catch block which is not good practice. From my point of view at least.

Reply:
ok